payments/providers/base.py

```diff
@@ -68,7 +89,7 @@ def ui_redirect_failure(cls, return_url: str, order: Order = None) -> HttpRespon
         This should be used after a failed payment instead of the
         standard Django redirect.
         """
-        return cls._redirect_to_ui(return_url, 'failure', order)
+        return cls._redirect_to_ui(return_url, 'failure', order) if return_url else HttpResponseServerError()
```

Comment:
Ditto, "Payment failure and cannot even redirect back to UI, damn oh damn."

Reply:
Yup, same thing.